### PR TITLE
Add durability policy changes to the summary

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description
This PR adds the durability policy changes to the summary of release-14 for the following PRs-
- https://github.com/vitessio/vitess/pull/10201
- https://github.com/vitessio/vitess/pull/10221
- https://github.com/vitessio/vitess/pull/10375
- https://github.com/vitessio/vitess/pull/10423

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: doc/releasenotes/14_0_0_summary.md [hunk 1]
```diff
@@ -195,3 +195,20 @@ Thanks to this rewriting we were previously able to plan sharded queries with an
...
+#### New and Augmented Commands
+The vtctld command `CreateKeyspace` has been augmented to take in an additional argument called `durability-policy` which will
+allow the users to set the correct durability policy at the creation of keyspaces.
+
+For existing keyspaces, a new command `SetKeyspaceDurabilityPolicy` has been added, which allows the users to change the
+durability policy of an existing keyspace.
+
+If semi-sync is not being used then durability policy should be set to `none` in the keyspace information. This is also the default option.
+
+If semi-sync is being used then durability policy should be set to `semi_sync` in the keyspace information and `-enable_semi_sync` should be set in vttablets.
```
Comment (deepthi) [inline doc/releasenotes/14_0_0_summary.md:214]:
```suggestion
If semi-sync is being used then durability policy should be set to `semi_sync` for the keyspace and `--enable_semi_sync` should be set on vttablets.
```

File: doc/releasenotes/14_0_0_summary.md [hunk 2]
```diff
@@ -195,3 +195,20 @@ Thanks to this rewriting we were previously able to plan sharded queries with an
 Queries with an `USING` condition that need to be sent to a sharded keyspace are no longer supported and will return an `unsupported` planner error.
 
 This change was made through pull request [#9767](https://github.com/vitessio/vitess/pull/9767).
+
+### Durability Policy
+
+#### Deprecation of durability_policy flag
+The flag `durability_policy` added to vtctl, vtctld and vtworker binaries has now been deprecated.
+Instead the durability policy for a given keyspace will be stored in the keyspace information in the topo-server.
```
[the remark this answers is not in the document]
Comment (GuptaManan100) [reply doc/releasenotes/14_0_0_summary.md:195]: Fixed

File: doc/releasenotes/14_0_0_summary.md [hunk 3]
```diff
@@ -195,3 +195,20 @@ Thanks to this rewriting we were previously able to plan sharded queries with an
...
+
+#### New and Augmented Commands
+The vtctld command `CreateKeyspace` has been augmented to take in an additional argument called `durability-policy` which will
+allow the users to set the correct durability policy at the creation of keyspaces.
```
[the remark this answers is not in the document]
Comment (GuptaManan100) [reply doc/releasenotes/14_0_0_summary.md]: Fixed

File: doc/releasenotes/14_0_0_summary.md [hunk 4]
```diff
@@ -195,3 +195,20 @@ Thanks to this rewriting we were previously able to plan sharded queries with an
...
+
+For existing keyspaces, a new command `SetKeyspaceDurabilityPolicy` has been added, which allows the users to change the
```
[the remark this answers is not in the document]
Comment (GuptaManan100) [reply doc/releasenotes/14_0_0_summary.md]: Fixed

File: doc/releasenotes/14_0_0_summary.md [hunk 5]
```diff
@@ -195,3 +195,20 @@ Thanks to this rewriting we were previously able to plan sharded queries with an
...
+durability policy of an existing keyspace.
+
+If semi-sync is not being used then durability policy should be set to `none` in the keyspace information. This is also the default option.
```
[the remark this answers is not in the document]
Comment (GuptaManan100) [reply doc/releasenotes/14_0_0_summary.md]: Fixed

File: doc/releasenotes/14_0_0_summary.md (hunk 1 above)
Comment (GuptaManan100) [reply doc/releasenotes/14_0_0_summary.md]: Fixed

File: doc/releasenotes/14_0_0_summary.md [hunk 6]
```diff
@@ -187,3 +187,28 @@ The throttler now checks in with the heartbeat writer to request heartbeats, any
 When `--heartbeat_on_demand_duration` is not set, there is no change in behavior.
 
 When `--heartbeat_on_demand_duration` is set to a positive value, then the throttler ensures that the heartbeat writer generated heartbeats for at least the following duration. This also means at the first throttler check, it's possible that heartbeats are idle, and so the first check will fail. As heartbeats start running, followup checks will get a more accurate lag evaluation and will respond accordingly. In a sense, it's a "cold engine" scenario, where the engine takes time to start up, and then runs smoothly.
+
+### Durability Policy
+
+#### Deprecation of durability_policy Flag
+The flag `durability_policy` added to vtctl, vtctld and vtworker binaries has now been deprecated.
```
Comment (ajm188) [inline doc/releasenotes/14_0_0_summary.md:194]: this wording is confusing to me; it _sounds_ like we just added the flag to vtctl, vtctld, and vtworker, and are also deprecating it.

suggestion:

> The durability policy for a keyspace is now stored in the keyspace record in the topo server.
> The `durability_policy` flag used by `vtctl`, `vtctld`, and `vtworker` binaries has been deprecated.

File: doc/releasenotes/14_0_0_summary.md [hunk 7]
```diff
@@ -187,3 +187,28 @@ The throttler now checks in with the heartbeat writer to request heartbeats, any
...
+#### New and Augmented Commands
+The vtctld command `CreateKeyspace` has been augmented to take in an additional argument called `durability-policy` which will
+allow users to set the desired durability policy for a keyspace at creation time.
+
+For existing keyspaces, a new command `SetKeyspaceDurabilityPolicy` has been added, which allows users to change the
+durability policy of an existing keyspace.
+
+If semi-sync is not being used then durability policy should be set to `none` for the keyspace. This is also the default option.
+
+If semi-sync is being used then durability policy should be set to `semi_sync` for the keyspace and `-enable_semi_sync` should be set on vttablets.
```
Comment (ajm188) [inline doc/releasenotes/14_0_0_summary.md:206]: `-enable_semi_sync` => `--enable_semi_sync`

File: doc/releasenotes/14_0_0_summary.md [hunk 8]
```diff
@@ -187,3 +187,28 @@ The throttler now checks in with the heartbeat writer to request heartbeats, any
...
+
+### Deprecation of Durability Configuration
+The `Durability` configuration is deprecated and removed from VTOrc. Instead VTOrc will find the durability policy of the keyspace from
+the topo server. This allows VTOrc to monitor and repair multiple keyspaces which have different durability policies in use.
+
+It will ignore the keyspaces which have no durability policy specified in the keyspace record. So on upgrading to v14, users must run
```
Comment (ajm188) [inline doc/releasenotes/14_0_0_summary.md:212]: "It will ignore" => "VTOrc will ignore"

File: doc/releasenotes/14_0_0_summary.md [hunk 9]
```diff
@@ -187,3 +187,28 @@ The throttler now checks in with the heartbeat writer to request heartbeats, any
...
+the command `SetKeyspaceDurabilityPolicy` specified above, to ensure VTOrc continues to work as desired.
```
Comment (ajm188) [inline doc/releasenotes/14_0_0_summary.md:213]: I'd put this in big bold, because **everyone** upgrading will need to do this.

**Review by ajm188 — CHANGES_REQUESTED** [review]

File: doc/releasenotes/14_0_0_summary.md (hunk 6 above)
Comment (GuptaManan100) [reply doc/releasenotes/14_0_0_summary.md]: Fixed

File: doc/releasenotes/14_0_0_summary.md (hunk 8 above)
Comment (GuptaManan100) [reply doc/releasenotes/14_0_0_summary.md]: Fixed

File: doc/releasenotes/14_0_0_summary.md (hunk 9 above)
Comment (GuptaManan100) [reply doc/releasenotes/14_0_0_summary.md]: Fixed

File: doc/releasenotes/14_0_0_summary.md (hunk 7 above)
Comment (GuptaManan100) [reply doc/releasenotes/14_0_0_summary.md]: Fixed

**Review by ajm188 — APPROVED** [review]

**Pull request merged**